src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONBuilder.java

```diff
@@ -171,39 +171,46 @@ private JSONBuilder writeCoordinates(CoordinateSequence coordinates) throws JSON
      * @param y X ordinate
      * @param z Z ordinate, can be {@code NaN}
      * @param m M ordinate, can be {@code NaN}
-     * @return the JSON builder instance, this allow chained calls
+     * @return the JSON builder instance, this allows chained calls
      */
     private JSONBuilder writeCoordinate(double x, double y, double z, double m) {
         // start encoding JSON array
         this.array();
         // adjust the order of X and Y ordinates if needed
         if (axisOrder == CRS.AxisOrder.NORTH_EAST) {
             // encode latitude first and then longitude
-            if (!Double.isNaN(y)) { // for 1d linear referencing cases
-                roundedValue(y);
-            }
-            roundedValue(x);
+            encodeCoordinateIfAvailable(y);
+            encodeCoordinateIfAvailable(x);
         } else {
             // encode longitude first and then latitude
-            roundedValue(x);
-            if (!Double.isNaN(y)) { // for 1d linear referencing cases
-                roundedValue(y);
-            }
+            encodeCoordinateIfAvailable(x);
+            encodeCoordinateIfAvailable(y);
         }
         // if Z value is not available but we have a measure, we set Z value to zero
         z = Double.isNaN(z) && !Double.isNaN(m) ? 0 : z;
         // encode Z value if available
-        if (!Double.isNaN(z)) {
-            roundedValue(z);
-        }
+        encodeCoordinateIfAvailable(z);
         // encode M value if available
-        if (!Double.isNaN(m)) {
-            roundedValue(m);
-        }
+        encodeCoordinateIfAvailable(m);
         // we are done with the array
         return this.endArray();
     }
 
+    private void encodeCoordinateIfAvailable(double value) {
```

Comment:
```suggestion
    private void encodeOrdinate(double value) {
```

Notes:
* This is encoding just a number. It's an "ordinate" not a coordinate. 
* The details of why and how encoding is performed can be provided in the method  javadoc instead. If we wanted to be correct, the name would otherwise be "encodeOrdinateIfAvailableAndInfinityAsString".

Reply:
Right, I fixed this [in the next commit](https://github.com/geoserver/geoserver/pull/7343/commits/916d424e20988f2ed9add4c46532b85162c5e985)